apps/minifront/src/state/swap/helpers.ts

```diff
@@ -63,3 +71,47 @@ export const sendCandlestickDataRequest = async (
     else throw err;
   }
 };
+
+const byBalanceDescending = (a: BalancesResponse, b: BalancesResponse) => {
+  const aExponent = getDisplayDenomExponentFromValueView(a.balanceView);
+  const bExponent = getDisplayDenomExponentFromValueView(b.balanceView);
+  const aAmount = fromBaseUnitAmount(getAmount(a.balanceView), aExponent);
+  const bAmount = fromBaseUnitAmount(getAmount(b.balanceView), bExponent);
+
+  return bAmount.comparedTo(aAmount);
+};
+
+const nonSwappableAssetPatterns = [
+  assetPatterns.lpNft,
+  assetPatterns.proposalNft,
+  assetPatterns.votingReceipt,
+  assetPatterns.auctionNft,
+  assetPatterns.lpNft,
+
+  // In theory, these asset types are swappable, but we have removed them for now to get a better UX
+  assetPatterns.delegationToken,
+  assetPatterns.unbondingToken,
+];
+
+const isSwappable = (metadata: Metadata) =>
+  nonSwappableAssetPatterns.every(pattern => !pattern.matches(getDisplay(metadata)));
+
+const isKnown = (balancesResponse: BalancesResponse) =>
+  getValueViewCaseFromBalancesResponse.optional()(balancesResponse) === 'knownAssetId';
+
+export const swappableBalancesResponsesSelector = (
```

Comment:
The helpers above these two selectors are simply moved over from `components/swap/helpers.ts`, but these two selectors are new. They allow us to use the `shared.balancesResponses` and `shared.assets` properties and filter/sort them for the swap feature.